src/resources/rmd/patch.R

```diff
@@ -40,58 +40,6 @@ if (requireNamespace("htmlwidgets", quietly = TRUE)) {
   assignInNamespace("resolveSizing", resolveSizing, ns = "htmlwidgets")
 }
 
-
-# override parse_block to assign chunk labels from yaml options
-knitr_parse_block <- knitr:::parse_block
-parse_block = function(code, header, params.src, markdown_mode = out_format('markdown')) {
-  originalParamsSrc <- params.src
-  engine = sub('^([a-zA-Z0-9_]+).*$', '\\1', params.src)
-  partitioned <- partition_yaml_options(engine, code)
-  params = sub('^([a-zA-Z0-9_]+)', '', params.src)
-  params <- knitr:::parse_params(params)
-  unnamed_label <- knitr::opts_knit$get('unnamed.chunk.label')
-  if (startsWith(params$label, unnamed_label)) {
-    label <- partitioned$yaml[["label"]] %||% partitioned$yaml[["id"]]
-    if (!is.null(label)) {
-      params.src <- sub("^[a-zA-Z0-9_]+ *[ ,]?", 
-                        paste0(engine, " ", label, ", "), 
-                        params.src)
-    }
-  } 
-  
-  # strip trailing comma and whitespace
-  params.src <- sub("\\s*,?\\s*$", "", params.src)
-  
-  # look for other options to forward. note that ideally we could extract *all*
-  # parameters and then pass partitioned$code below, however we can construct
-  # cases where deparsed versions of the options include a newline, which causes
-  # an error. we'll wait and see if this capability is incorporated natively
-  # into knitr parse_block -- if it's not then we can pursue more robust versions
-  # of textual option forwarding that don't run into newlines 
```

Comment:
@jjallaire I guess this part is no longer necessary, since I parse these YAML options and merge them right away here: https://github.com/yihui/knitr/blob/c9b7e3acae3248e5a48beddd49e98d37d1cf1517/R/parser.R#L96-L97 (so you don't need to do it in `knitr_options_hook()` in `src/resources/rmd/hooks.R` above) Not sure if I missed or misunderstood anything.